core/backend/src/ECSqlStatement.ts

```diff
@@ -332,46 +336,43 @@ export class ECSqlStatement implements IterableIterator<any>, IDisposable {
    * - [Code Samples]($docs/learning/backend/ECSQLCodeExamples#working-with-the-query-result)
    */
   public getRow(args?: ECSqlRowArg): any {
-    args = args ?? {};
+    if (!this._stmt)
+      throw new Error("ECSqlStatement is not prepared");
 
-    // Set default rowFormat if not provided
+    args = args ?? {};
     if (args.rowFormat === undefined) {
       args.rowFormat = QueryRowFormat.UseJsPropertyNames;
     }
-
-    const toRowOptions = {
-      classIdsToClassNames: true,
-      rowFormat: args.rowFormat,
-    };
-    const resp = this._stmt!.toRow(toRowOptions); // eslint-disable-line @typescript-eslint/no-non-null-assertion
+    if (args.classIdsToClassNames === undefined) {
+      args.classIdsToClassNames = true;
+    }
+    const resp = this._stmt.toRow({
+      classIdsToClassNames: args.classIdsToClassNames,
+      useJsName: args.rowFormat === QueryRowFormat.UseJsPropertyNames,
+      abbreviateBlobs: false,
+      // on 5.x this should be removed or set to false. It is only here to maintain compatibility with 4.x
+      doNotConvertClassIdsToClassNamesWhenAliased: true,
+    });
     return this.formatCurrentRow(resp, args.rowFormat);
   }
 
-  /**
-   * @internal
-   */
-  public formatCurrentRow(currentResp: any, rowFormat: QueryRowFormat = QueryRowFormat.UseJsPropertyNames): any[] | object {
+  private formatCurrentRow(currentResp: any, rowFormat: QueryRowFormat = QueryRowFormat.UseJsPropertyNames): any[] | object {
+    if (!this._stmt)
+      throw new Error("ECSqlStatement is not prepared");
+
+    if (rowFormat === QueryRowFormat.UseECSqlPropertyIndexes)
+      return currentResp.data;
+
     if (this._props.length === 0) {
-      const resp = this._stmt!.getMetadata(); // eslint-disable-line @typescript-eslint/no-non-null-assertion
+      const resp = this._stmt.getMetadata();
       this._props = new PropertyMetaDataMap(resp.meta);
     }
     const formattedRow = {};
-    const uniqueNames = new Map<string, number>();
     for (const prop of this._props) {
       const propName = rowFormat === QueryRowFormat.UseJsPropertyNames ? prop.jsonName : prop.name;
       const val = currentResp.data[prop.index];
       if (typeof val !== "undefined" && val !== null) {
-        let uniquePropName = propName;
```

Comment:
Is this snippet no longer necessary because you're checking for uniqueECSqlMembers in EcSqlRowAdaptor::GetMetaData on the native side now?